closed/make/Docs.gmk

```diff
@@ -20,4 +20,115 @@
+OpenJ9_GROUP_MODULES := openj9.*
+OpenJ9_GROUP_DESCRIPTION := \
+    Modules whose names start with {@code openj9}. contain APIs provided by the \
+    <a href="$(OPENJ9_JAVADOC_BASE_URL)">Eclipse OpenJ9</a> project. \
+    These APIs can only be used with a JDK which includes the OpenJ9 virtual machine. \
+    #
+JDK_GROUPS += OpenJ9
+
+################################################################################
+# In the jdk.management module:
+# 1. Some openj9 specific packages are added
+# 2. Some classes in the com.sun.management package are reimplemented in an
+#    openj9 specific package
+# 3. Some classes in the com.sun.management package are not relevant (cannot be
+#    used) with an openj9 based jdk.
+# The code below renames the files which cannot be used while javadoc is
+# executed so no javadoc is generated for those classes.
+#
+# This is not done within a target because the javadoc dependencies are generated
+# from the source tree before executing any targets, so dependencies would be generated
+# for files found before they are renamed, causing build errors.  
+
+OPENJDK_SUN_MGMT_SRCDIR = $(OPENJDK_JCL_TOPDIR)/jdk.management/share/classes/jdk.management/com/sun/management
+OPENJ9_SUN_MGMT_SRCDIR = $(OPENJ9_JCL_TOPDIR)/jdk.management/share/classes/com/sun/management
+
+JDK_API_CUSTOM_TARGETS += openj9-docs-set-javadoc-bottom openj9-docs-restore-files
+
+# Code should only be executed when generating the docs-jdk-api-javadoc target.
+# If executed for any other targets the renames would be done but the target
+# to restore the files would not be executed.
+
+ifneq (,$(findstring docs-jdk-api-javadoc,$(MAKECMDGOALS)))
+  # First rename the files we want to generate javadoc for, and to keep package-info.java which may be present from a previous run.
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.java $(OPENJDK_SUN_MGMT_SRCDIR)/GarbageCollectorMXBean.javaz; fi)
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/OperatingSystemMXBean.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/OperatingSystemMXBean.java $(OPENJDK_SUN_MGMT_SRCDIR)/OperatingSystemMXBean.javaz; fi)
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.java $(OPENJDK_SUN_MGMT_SRCDIR)/UnixOperatingSystemMXBean.javaz; fi)
+  CMD_OUTPUT = $(shell if [ -f $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.java ]; then $(MV) $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.java $(OPENJDK_SUN_MGMT_SRCDIR)/package-info.javaz; fi)
+  # Then rename any remaining files we don't want to generate javadoc for
+  # This finds all files ending in .java and then renames each one with an 'x' added add its last character.
+  CMD_OUTPUT = $(shell find $(OPENJDK_SUN_MGMT_SRCDIR) -maxdepth 1 -type 'f' -name "*.java" | xargs -I {} sh -c 'set -x; $(MV) $$1 $${1}x; set +x' -- {})
```

Comment:
Please remove unnecessary quotes (e.g. `'f'`, `"/openj9"`, `".html"`).

Reply:
Will change 'f' to f.

Reply:
Changed (now in closed/make/Docs-post.gmk).